.travis.yml

```diff
@@ -1,21 +1,36 @@
+os:
+  - linux
+  - windows
 language: go
 go:
-- 1.14.x
-- 1.13.x
-
+  - 1.13.x
+  - 1.14.x
+git:
+  depth: false
+env:
+  - V=
+  - V=v2
 install:
-- go get -t ./...
-- go get github.com/mattn/goveralls
-- go get github.com/wadey/gocovmerge
-- go get -u honnef.co/go/tools/cmd/staticcheck
-- go get -u github.com/client9/misspell/cmd/misspell
-
+  - go get -t ./...
+  - go get -u honnef.co/go/tools/cmd/staticcheck
+  - go get -u github.com/client9/misspell/cmd/misspell
+  - go get github.com/mattn/goveralls
+  - go get github.com/wadey/gocovmerge
 before_script:
-- $(exit $(go fmt ./... | wc -l))
-- go vet ./...
-- misspell -error -locale US .
-- staticcheck ./...
-
+  - cd $TRAVIS_BUILD_DIR/${V}
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then EXCLUDE_VENDOR=$(go list ./... | grep -v "/vendor/") && $(exit $(go fmt $EXCLUDE_VENDOR | wc -l)) && go vet $EXCLUDE_VENDOR; fi
+  - go vet ./...
+  - misspell -error -locale US .
+  - staticcheck ./...
 script:
-- go test -v -race ./...
-- if [[ "$TRAVIS_GO_VERSION" =~ 1.13 ]]; then ./scripts/cov.sh TRAVIS; fi
+  - go test -v -race ./...
+  - go test -v -coverprofile=./coverage.out ./...
```

Comment:
If you are going to report coverage only once, you may want to simply not run it here for matrixes that do not report. Say you report only for Go 1.14 and v2, then you should run only then. That being said, jwt tests are run quickly, so that is less of a concern.

Reply:
Not quite sure what the coverage report means for the multi version repo.

Reply:
That's my point, we should do the coverage only once per PR. Here it will run for all the matrix entries, but we should post only once.

Reply:
Again, why run the coverage if you are not going to report it? Since down below we push it on v2, for linux, and Go 1.14, I would have run the coverage code only in those conditions.